extensions/wikia/SiteWideMessages/SpecialSiteWideMessages_body.php

```diff
@@ -300,6 +300,13 @@ function execute($subpage) {
 	}
 
 	//DB functions
+
+	/**
+	 * @param User $mSender
+	 * @param $mText
+	 * @param $formData
+	 * @return array
+	 */
 	private function sendMessage( $mSender, $mText, $formData ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this function "sendMessage" is 106 which is greater than 20 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)

[project-id]: wikia:app
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This function "sendMessage" has 586 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)

[project-id]: wikia:app